From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ricardo Pardini <ricardo@pardini.net>
Date: Sun, 18 Feb 2024 14:33:37 +0100
Subject: boot_rkimg: don't try rockchip_u2phy_vbus_detect when RECOVERY button
 pressed

- boot_rkimg: add debugs for download button not working
---
 arch/arm/mach-rockchip/boot_rkimg.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-rockchip/boot_rkimg.c b/arch/arm/mach-rockchip/boot_rkimg.c
index 9e6c3556a9e2..f79aed467bc1 100644
--- a/arch/arm/mach-rockchip/boot_rkimg.c
+++ b/arch/arm/mach-rockchip/boot_rkimg.c
@@ -322,14 +322,17 @@ void setup_download_mode(void)
 
 	boot_devtype_init();
 
+	printf("setup_download_mode checking if download %skey pressed...\n",
+	       is_hotkey(HK_ROCKUSB_DNL) ? "is HOT YES" : "is NOT HOT");
+	
 	if (rockchip_dnl_key_pressed() || is_hotkey(HK_ROCKUSB_DNL)) {
 		printf("download %skey pressed... ",
 		       is_hotkey(HK_ROCKUSB_DNL) ? "hot" : "");
 #ifdef CONFIG_CMD_ROCKUSB
-		vbus = rockchip_u2phy_vbus_detect();
+		// vbus = rockchip_u2phy_vbus_detect(); // rpardini: don't do this, it fails with -19
 #endif
 		if (vbus > 0) {
-			printf("%sentering download mode...\n",
+			printf("%sentering download mode vbus > 0...\n",
 			       IS_ENABLED(CONFIG_CMD_ROCKUSB) ?
 			       "" : "no rockusb, ");
 
@@ -339,7 +342,7 @@ void setup_download_mode(void)
 			if ((fdt_node_offset_by_compatible(blob, -1, "radxa,rockpie")) >= 0) {
 				run_command("download", 0);
 			}
-			printf("entering recovery mode!\n");
+			printf("entering recovery mode vbus not > 0!\n");
 			env_set("reboot_mode", "recovery-key");
 		}
 	} else if (is_hotkey(HK_FASTBOOT)) {
-- 
Armbian

